### ISSUE TITLE
Bug: Somehow option is shifted (editor.getOption does not work as expected)

### ISSUE THREAD

**Issue by LumaKernel** [suren-atoyan/monaco-react#546]
**Describe the bug**
`editor.getOption` can be lookuped by index number of [option enums](https://github.com/microsoft/vscode/blob/0e925d67006d018effb840fd533acd01d61b5e4a/src/vs/monaco.d.ts#L4708).

**To Reproduce**

1. Go to codesandbox this example: https://codesandbox.io/p/sandbox/inspiring-dubinsky-y7rwnc
```tsx
import { editor as monacoEditor } from "monaco-editor";
import Editor, { OnMount } from "@monaco-editor/react";
import { useState } from "react";

function App() {
  const [json, setJson] = useState("...");
  const handleMount: OnMount = (editor) => {
    setJson(
      JSON.stringify(
        Object.fromEntries(
          // eslint-disable-next-line @typescript-eslint/no-explicit-any
          (editor.getOptions() as unknown as { _values: any[] })._values.map(
            (v, i) => [monacoEditor.EditorOption[i], v],
          ),
        ),
        null,
        2,
      ),
    );
  };
  return (
    <div>
      <pre>
        <code>{json}</code>
      </pre>
      <Editor onMount={handleMount} />
    </div>
  );
}

export default App;
```
3. See JSON

```json
{
  "acceptSuggestionOnCommitCharacter": true,
  "acceptSuggestionOnEnter": "on",
  "accessibilitySupport": 0,
  "accessibilityPageSize": 10,
  "ariaLabel": "Editor content;Press Alt+F1 for Accessibility Options.",
  "ariaRequired": false,
  "autoClosingBrackets": "languageDefined",
  "autoClosingComments": true,
  "screenReaderAnnounceInlineSuggestion": "auto",
  "autoClosingDelete": "auto",
  "autoClosingOvertype": "languageDefined",
  "autoClosingQuotes": 3,
  "autoIndent": true,
  "automaticLayout": "languageDefined",
  "autoSurround": {
    "enabled": true,
    "independentColorPoolPerBracketType": false
  },
  "bracketPairColorization": {
    "bracketPairs": false,
    "bracketPairsHorizontal": "active",
    "highlightActiveBracketPair": true,
    "indentation": true,
    "highlightActiveIndentation": true
  },
  "guides": true,
  "codeLens": "",
  "codeLensFontFamily": 0,
  "codeLensFontSize": true,
  "colorDecorators": 500,
  "colorDecoratorsLimit": false,
  "columnSelection": {
    "insertSpace": true,
    "ignoreEmptyLines": true
  },
  "comments": true,
  "contextmenu": true,
  "copyWithSyntaxHighlighting": 1,
  "cursorBlinking": "off",
  "cursorSmoothCaretAnimation": 1,
  "cursorStyle": 0,
  "cursorSurroundingLines": "default",
  "cursorSurroundingLinesStyle": 0,
  "cursorWidth": false,
  "disableLayerHinting": false,
  "disableMonospaceOptimizations": false,
  "domReadOnly": true,
  "dragAndDrop": {
    "enabled": true,
    "showDropSelector": "afterDrop"
  },
  "dropIntoEditor": true,
  "emptySelectionClipboard": "svg",
  "experimentalWhitespaceRendering": "",
  "extraEditorClassName": 5,
  "fastScrollSensitivity": {
    "cursorMoveOnType": true,
    "seedSearchStringFromSelection": "always",
    "autoFindInSelection": "never",
    "globalFindClipboard": false,
    "addExtraSpaceOnTop": true,
    "loop": true
  },
  "find": false,
  "fixedOverflowWidgets": true,
  "folding": "auto",
  "foldingStrategy": true,
  "foldingHighlight": false,
  "foldingImportsByDefault": 5000,
  "foldingMaximumRegions": false,
  "unfoldOnClickAfterEndOfLine": "Menlo, Monaco, 'Courier New', monospace",
  "fontFamily": {
    "pixelRatio": 2,
    "fontFamily": "Menlo, Monaco, 'Courier New', monospace",
    "fontWeight": "normal",
    "fontSize": 12,
    "fontFeatureSettings": "\"liga\" off, \"calt\" off",
    "fontVariationSettings": "normal",
    "lineHeight": 18,
    "letterSpacing": 0,
    "version": 2,
    "isTrusted": true,
    "isMonospace": true,
    "typicalHalfwidthCharacterWidth": 7.2265625,
    "typicalFullwidthCharacterWidth": 12,
    "canUseHalfwidthRightwardsArrow": false,
    "spaceWidth": 7.2265625,
    "middotWidth": 7.2265625,
    "wsmiddotWidth": 3.28125,
    "maxDigitWidth": 7.2265625
  },
  "fontInfo": "\"liga\" off, \"calt\" off",
  "fontLigatures": 12,
  "fontSize": "normal",
  "fontWeight": "normal",
  "fontVariations": false,
  "formatOnPaste": false,
  "formatOnType": false,
  "glyphMargin": {
    "multiple": "peek",
    "multipleDefinitions": "peek",
    "multipleTypeDefinitions": "peek",
    "multipleDeclarations": "peek",
    "multipleImplementations": "peek",
    "multipleReferences": "peek",
    "alternativeDefinitionCommand": "editor.action.goToReferences",
    "alternativeTypeDefinitionCommand": "editor.action.goToReferences",
    "alternativeDeclarationCommand": "editor.action.goToReferences",
    "alternativeImplementationCommand": "",
    "alternativeReferenceCommand": ""
  },
  "gotoLocation": false,
  "hideCursorInOverviewRuler": {
    "enabled": true,
    "delay": 300,
    "sticky": true,
    "above": true
  },
  "hover": false,
  "inDiffEditor": {
    "enabled": true,
    "mode": "subwordSmart",
    "showToolbar": "onHover",
    "suppressSuggestions": false,
    "keepOnBlur": false
  },
  "inlineSuggest": 0,
  "letterSpacing": {
    "enabled": true
  },
  "lightbulb": 10,
  "lineDecorationsWidth": 18,
  "lineHeight": {
    "renderType": 1,
    "renderFn": null
  },
  "lineNumbers": 5,
  "lineNumbersMinChars": false,
  "linkedEditing": true,
  "links": "always",
  "matchBrackets": {
    "enabled": true,
    "size": "proportional",
    "side": "right",
    "showSlider": "mouseover",
    "autohide": false,
    "renderCharacters": true,
    "maxColumn": 120,
    "scale": 1
  },
  "minimap": "text",
  "mouseStyle": 1,
  "mouseWheelScrollSensitivity": false,
  "mouseWheelZoom": true,
  "multiCursorMergeOverlapping": "altKey",
  "multiCursorModifier": "spread",
  "multiCursorPaste": 10000,
  "multiCursorLimit": true,
  "occurrencesHighlight": true,
  "overviewRulerBorder": 2,
  "overviewRulerLanes": {
    "top": 0,
    "bottom": 0
  },
  "padding": {
    "enabled": true,
    "showPasteSelector": "afterPaste"
  },
  "pasteAs": {
    "enabled": true,
    "cycle": true
  },
  "parameterHints": "tree",
  "peekWidgetDefaultFocus": false,
  "definitionLinkOpensInPeek": {
    "other": "on",
    "comments": "off",
    "strings": "off"
  },
  "quickSuggestions": 10,
  "quickSuggestionsDelay": false,
  "readOnlyMessage": false,
  "renameOnType": true,
  "renderControlCharacters": "on",
  "renderFinalNewline": "line",
  "renderLineHighlight": false,
  "renderLineHighlightOnlyWhenFocus": "editable",
  "renderValidationDecorations": "selection",
  "renderWhitespace": 15,
  "revealHorizontalRightPadding": true,
  "roundedSelection": [],
  "rulers": {
    "vertical": 1,
    "horizontal": 1,
    "arrowSize": 11,
    "useShadows": true,
    "verticalHasArrows": false,
    "horizontalHasArrows": false,
    "horizontalScrollbarSize": 12,
    "horizontalSliderSize": 12,
    "verticalScrollbarSize": 14,
    "verticalSliderSize": 14,
    "handleMouseWheel": true,
    "alwaysConsumeMouseWheel": true,
    "scrollByPage": false
  },
  "scrollbar": 4,
  "scrollBeyondLastColumn": true,
  "scrollBeyondLastLine": true,
  "scrollPredominantAxis": true,
  "selectionClipboard": true,
  "selectionHighlight": true,
  "selectOnLineNumbers": "mouseover",
  "showFoldingControls": true,
  "showUnused": "inline",
  "snippetSuggestions": {
    "selectLeadingAndTrailingWhitespace": true,
    "selectSubwords": true
  },
  "smartSelect": false,
  "smoothScrolling": {
    "enabled": false,
    "maxLineCount": 5,
    "defaultModel": "outlineModel",
    "scrollWithEditor": true
  },
  "stickyScroll": false,
  "stickyTabStops": 10000,
  "stopRenderingLineAfter": {
    "insertMode": "insert",
    "filterGraceful": true,
    "snippetsPreventQuickSuggestions": false,
    "localityBonus": false,
    "shareSuggestSelections": false,
    "selectionMode": "always",
    "showIcons": true,
    "showStatusBar": false,
    "preview": false,
    "previewMode": "subwordSmart",
    "showInlineDetails": true,
    "showMethods": true,
    "showFunctions": true,
    "showConstructors": true,
    "showDeprecated": true,
    "matchOnWordStartOnly": true,
    "showFields": true,
    "showVariables": true,
    "showClasses": true,
    "showStructs": true,
    "showInterfaces": true,
    "showModules": true,
    "showProperties": true,
    "showEvents": true,
    "showOperators": true,
    "showUnits": true,
    "showValues": true,
    "showConstants": true,
    "showEnums": true,
    "showEnumMembers": true,
    "showKeywords": true,
    "showWords": true,
    "showColors": true,
    "showFiles": true,
    "showReferences": true,
    "showFolders": true,
    "showTypeParameters": true,
    "showSnippets": true,
    "showUsers": true,
    "showIssues": true
  },
  "suggest": 0,
  "suggestFontSize": 0,
  "suggestLineHeight": true,
  "suggestOnTriggerCharacters": "first",
  "suggestSelection": "off",
  "tabCompletion": 0,
  "tabIndex": {
    "nonBasicASCII": "inUntrustedWorkspace",
    "invisibleCharacters": true,
    "ambiguousCharacters": true,
    "includeComments": "inUntrustedWorkspace",
    "includeStrings": true,
    "allowedCharacters": {},
    "allowedLocales": {
      "_os": true,
      "_vscode": true
    }
  },
  "unicodeHighlighting": "prompt",
  "unusualLineTerminators": true,
  "useShadowDOM": true,
  "useTabStops": "normal",
  "wordBreak": "`~!@#$%^&*()-=+[{]}\\|;:'\",.<>/?",
  "wordSeparators": "off",
  "wordWrap": " \t})]?|/&.,;¢°′″‰℃、。｡､￠，．：；？！％・･ゝゞヽヾーァィゥェォッャュョヮヵヶぁぃぅぇぉっゃゅょゎゕゖㇰㇱㇲㇳㇴㇵㇶㇷㇸㇹㇺㇻㇼㇽㇾㇿ々〻ｧｨｩｪｫｬｭｮｯｰ”〉》」』】〕）］｝｣",
  "wordWrapBreakAfterCharacters": "([{‘“〈《「『【〔（［｛｢£¥＄￡￥+＋",
  "wordWrapBreakBeforeCharacters": 80,
  "wordWrapColumn": "inherit",
  "wordWrapOverride1": "inherit",
  "wordWrapOverride2": 1,
  "wrappingIndent": "simple",
  "wrappingStrategy": true,
  "showDeprecated": {
    "enabled": "on",
    "fontSize": 0,
    "fontFamily": "",
    "padding": false
  },
  "inlayHints": "monaco-editor no-user-select mac  showUnused showDeprecated",
  "editorClassName": 2,
  "pixelRatio": false,
  "tabFocusMode": {
    "width": 5,
    "height": 5,
    "glyphMarginLeft": 0,
    "glyphMarginWidth": 0,
    "glyphMarginDecorationLaneCount": 1,
    "lineNumbersLeft": 0,
    "lineNumbersWidth": 36,
    "decorationsLeft": 36,
    "decorationsWidth": 26,
    "contentLeft": 62,
    "contentWidth": -65,
    "minimap": {
      "renderMinimap": 1,
      "minimapLeft": -17,
      "minimapWidth": 8,
      "minimapHeightIsEditorHeight": false,
      "minimapIsSampling": false,
      "minimapScale": 2,
      "minimapLineHeight": 4,
      "minimapCanvasInnerWidth": 16,
      "minimapCanvasInnerHeight": 10,
      "minimapCanvasOuterWidth": 8,
      "minimapCanvasOuterHeight": 5
    },
    "viewportColumn": 1,
    "isWordWrapMinified": false,
    "isViewportWrapping": false,
    "wrappingColumn": -1,
    "verticalScrollbarWidth": 14,
    "horizontalScrollbarHeight": 12,
    "overviewRuler": {
      "top": 0,
      "width": 14,
      "height": 5,
      "right": 0
    }
  },
  "layoutInfo": {
    "isDominatedByLongLines": false,
    "isWordWrapMinified": false,
    "isViewportWrapping": false,
    "wrappingColumn": -1
  },
  "wrappingInfo": false,
  "defaultColorDecorators": "clickAndHover",
  "colorDecoratorsActivatedOn": false
}
```

You may see for example, `*FontSize` with non number values, `fontInfo` with string (actually object).

**Expected behavior**

This option JSON is got by [react-monaco-editor](https://github.com/react-monaco-editor/react-monaco-editor) with the same version of monaco-editor (0.44.0)

```json
{
  "acceptSuggestionOnCommitCharacter": true,
  "acceptSuggestionOnEnter": "on",
  "accessibilitySupport": 0,
  "accessibilityPageSize": 10,
  "ariaLabel": "Editor content;Press Alt+F1 for Accessibility Options.",
  "ariaRequired": false,
  "autoClosingBrackets": "languageDefined",
  "autoClosingComments": "languageDefined",
  "screenReaderAnnounceInlineSuggestion": true,
  "autoClosingDelete": "auto",
  "autoClosingOvertype": "auto",
  "autoClosingQuotes": "languageDefined",
  "autoIndent": 3,
  "automaticLayout": false,
  "autoSurround": "languageDefined",
  "bracketPairColorization": {
    "enabled": true,
    "independentColorPoolPerBracketType": false
  },
  "guides": {
    "bracketPairs": false,
    "bracketPairsHorizontal": "active",
    "highlightActiveBracketPair": true,
    "indentation": true,
    "highlightActiveIndentation": true
  },
  "codeLens": true,
  "codeLensFontFamily": "",
  "codeLensFontSize": 0,
  "colorDecorators": true,
  "colorDecoratorsLimit": 500,
  "columnSelection": false,
  "comments": {
    "insertSpace": true,
    "ignoreEmptyLines": true
  },
  "contextmenu": true,
  "copyWithSyntaxHighlighting": true,
  "cursorBlinking": 1,
  "cursorSmoothCaretAnimation": "off",
  "cursorStyle": 1,
  "cursorSurroundingLines": 0,
  "cursorSurroundingLinesStyle": "default",
  "cursorWidth": 0,
  "disableLayerHinting": false,
  "disableMonospaceOptimizations": false,
  "domReadOnly": false,
  "dragAndDrop": true,
  "dropIntoEditor": {
    "enabled": true,
    "showDropSelector": "afterDrop"
  },
  "emptySelectionClipboard": true,
  "experimentalWhitespaceRendering": "svg",
  "extraEditorClassName": "",
  "fastScrollSensitivity": 5,
  "find": {
    "cursorMoveOnType": true,
    "seedSearchStringFromSelection": "always",
    "autoFindInSelection": "never",
    "globalFindClipboard": false,
    "addExtraSpaceOnTop": true,
    "loop": true
  },
  "fixedOverflowWidgets": false,
  "folding": true,
  "foldingStrategy": "auto",
  "foldingHighlight": true,
  "foldingImportsByDefault": false,
  "foldingMaximumRegions": 5000,
  "unfoldOnClickAfterEndOfLine": false,
  "fontFamily": "Menlo, Monaco, 'Courier New', monospace",
  "fontInfo": {
    "pixelRatio": 2.5,
    "fontFamily": "Menlo, Monaco, 'Courier New', monospace",
    "fontWeight": "normal",
    "fontSize": 12,
    "fontFeatureSettings": "\"liga\" off, \"calt\" off",
    "fontVariationSettings": "normal",
    "lineHeight": 18,
    "letterSpacing": 0,
    "version": 2,
    "isTrusted": true,
    "isMonospace": true,
    "typicalHalfwidthCharacterWidth": 7.2265625,
    "typicalFullwidthCharacterWidth": 12,
    "canUseHalfwidthRightwardsArrow": false,
    "spaceWidth": 7.2265625,
    "middotWidth": 7.2265625,
    "wsmiddotWidth": 3.28125,
    "maxDigitWidth": 7.2265625
  },
  "fontLigatures": "\"liga\" off, \"calt\" off",
  "fontSize": 12,
  "fontWeight": "normal",
  "fontVariations": "normal",
  "formatOnPaste": false,
  "formatOnType": false,
  "glyphMargin": false,
  "gotoLocation": {
    "multiple": "peek",
    "multipleDefinitions": "peek",
    "multipleTypeDefinitions": "peek",
    "multipleDeclarations": "peek",
    "multipleImplementations": "peek",
    "multipleReferences": "peek",
    "alternativeDefinitionCommand": "editor.action.goToReferences",
    "alternativeTypeDefinitionCommand": "editor.action.goToReferences",
    "alternativeDeclarationCommand": "editor.action.goToReferences",
    "alternativeImplementationCommand": "",
    "alternativeReferenceCommand": ""
  },
  "hideCursorInOverviewRuler": false,
  "hover": {
    "enabled": true,
    "delay": 300,
    "hidingDelay": 300,
    "sticky": true,
    "above": true
  },
  "inDiffEditor": false,
  "inlineSuggest": {
    "enabled": true,
    "mode": "subwordSmart",
    "showToolbar": "onHover",
    "suppressSuggestions": false,
    "keepOnBlur": false
  },
  "letterSpacing": 0,
  "lightbulb": {
    "enabled": true
  },
  "lineDecorationsWidth": 10,
  "lineHeight": 18,
  "lineNumbers": {
    "renderType": 1,
    "renderFn": null
  },
  "lineNumbersMinChars": 5,
  "linkedEditing": false,
  "links": true,
  "matchBrackets": "always",
  "minimap": {
    "enabled": true,
    "size": "proportional",
    "side": "right",
    "showSlider": "mouseover",
    "autohide": false,
    "renderCharacters": true,
    "maxColumn": 120,
    "scale": 1
  },
  "mouseStyle": "text",
  "mouseWheelScrollSensitivity": 1,
  "mouseWheelZoom": false,
  "multiCursorMergeOverlapping": true,
  "multiCursorModifier": "altKey",
  "multiCursorPaste": "spread",
  "multiCursorLimit": 10000,
  "occurrencesHighlight": true,
  "overviewRulerBorder": true,
  "overviewRulerLanes": 2,
  "padding": {
    "top": 0,
    "bottom": 0
  },
  "pasteAs": {
    "enabled": true,
    "showPasteSelector": "afterPaste"
  },
  "parameterHints": {
    "enabled": true,
    "cycle": true
  },
  "peekWidgetDefaultFocus": "tree",
  "definitionLinkOpensInPeek": false,
  "quickSuggestions": {
    "other": "on",
    "comments": "off",
    "strings": "off"
  },
  "quickSuggestionsDelay": 10,
  "readOnly": true,
  "renameOnType": false,
  "renderControlCharacters": true,
  "renderFinalNewline": "on",
  "renderLineHighlight": "line",
  "renderLineHighlightOnlyWhenFocus": false,
  "renderValidationDecorations": "editable",
  "renderWhitespace": "boundary",
  "revealHorizontalRightPadding": 15,
  "roundedSelection": true,
  "rulers": [],
  "scrollbar": {
    "vertical": 1,
    "horizontal": 1,
    "arrowSize": 11,
    "useShadows": true,
    "verticalHasArrows": false,
    "horizontalHasArrows": false,
    "horizontalScrollbarSize": 12,
    "horizontalSliderSize": 12,
    "verticalScrollbarSize": 14,
    "verticalSliderSize": 14,
    "handleMouseWheel": true,
    "alwaysConsumeMouseWheel": true,
    "scrollByPage": false
  },
  "scrollBeyondLastColumn": 4,
  "scrollBeyondLastLine": true,
  "scrollPredominantAxis": true,
  "selectionClipboard": true,
  "selectionHighlight": true,
  "selectOnLineNumbers": true,
  "showFoldingControls": "mouseover",
  "showUnused": true,
  "snippetSuggestions": "inline",
  "smartSelect": {
    "selectLeadingAndTrailingWhitespace": true,
    "selectSubwords": true
  },
  "smoothScrolling": false,
  "stickyScroll": {
    "enabled": false,
    "maxLineCount": 5,
    "defaultModel": "outlineModel",
    "scrollWithEditor": true
  },
  "stickyTabStops": false,
  "stopRenderingLineAfter": 10000,
  "suggest": {
    "insertMode": "insert",
    "filterGraceful": true,
    "snippetsPreventQuickSuggestions": false,
    "localityBonus": false,
    "shareSuggestSelections": false,
    "selectionMode": "always",
    "showIcons": true,
    "showStatusBar": false,
    "preview": false,
    "previewMode": "subwordSmart",
    "showInlineDetails": true,
    "showMethods": true,
    "showFunctions": true,
    "showConstructors": true,
    "showDeprecated": true,
    "matchOnWordStartOnly": true,
    "showFields": true,
    "showVariables": true,
    "showClasses": true,
    "showStructs": true,
    "showInterfaces": true,
    "showModules": true,
    "showProperties": true,
    "showEvents": true,
    "showOperators": true,
    "showUnits": true,
    "showValues": true,
    "showConstants": true,
    "showEnums": true,
    "showEnumMembers": true,
    "showKeywords": true,
    "showWords": true,
    "showColors": true,
    "showFiles": true,
    "showReferences": true,
    "showFolders": true,
    "showTypeParameters": true,
    "showSnippets": true,
    "showUsers": true,
    "showIssues": true
  },
  "suggestFontSize": 0,
  "suggestLineHeight": 0,
  "suggestOnTriggerCharacters": true,
  "suggestSelection": "first",
  "tabCompletion": "off",
  "tabIndex": 0,
  "unicodeHighlighting": {
    "nonBasicASCII": "inUntrustedWorkspace",
    "invisibleCharacters": true,
    "ambiguousCharacters": true,
    "includeComments": "inUntrustedWorkspace",
    "includeStrings": true,
    "allowedCharacters": {},
    "allowedLocales": {
      "_os": true,
      "_vscode": true
    }
  },
  "unusualLineTerminators": "prompt",
  "useShadowDOM": true,
  "useTabStops": true,
  "wordBreak": "normal",
  "wordSeparators": "`~!@#$%^&*()-=+[{]}\\|;:'\",.<>/?",
  "wordWrap": "off",
  "wordWrapBreakAfterCharacters": " \t})]?|/&.,;¢°′″‰℃、。｡､￠，．：；？！％・･ゝゞヽヾーァィゥェォッャュョヮヵヶぁぃぅぇぉっゃゅょゎゕゖㇰㇱㇲㇳㇴㇵㇶㇷㇸㇹㇺㇻㇼㇽㇾㇿ々〻ｧｨｩｪｫｬｭｮｯｰ”〉》」』】〕）］｝｣",
  "wordWrapBreakBeforeCharacters": "([{‘“〈《「『【〔（［｛｢£¥＄￡￥+＋",
  "wordWrapColumn": 80,
  "wordWrapOverride1": "inherit",
  "wordWrapOverride2": "inherit",
  "wrappingIndent": 1,
  "wrappingStrategy": "simple",
  "showDeprecated": true,
  "inlayHints": {
    "enabled": "on",
    "fontSize": 0,
    "fontFamily": "",
    "padding": false
  },
  "editorClassName": "monaco-editor no-user-select mac  showUnused showDeprecated",
  "pixelRatio": 2.5,
  "tabFocusMode": false,
  "layoutInfo": {
    "width": 597,
    "height": 400,
    "glyphMarginLeft": 0,
    "glyphMarginWidth": 0,
    "glyphMarginDecorationLaneCount": 1,
    "lineNumbersLeft": 0,
    "lineNumbersWidth": 36,
    "decorationsLeft": 36,
    "decorationsWidth": 26,
    "contentLeft": 62,
    "contentWidth": 476,
    "minimap": {
      "renderMinimap": 1,
      "minimapLeft": 524,
      "minimapWidth": 59,
      "minimapHeightIsEditorHeight": false,
      "minimapIsSampling": false,
      "minimapScale": 2,
      "minimapLineHeight": 4,
      "minimapCanvasInnerWidth": 147,
      "minimapCanvasInnerHeight": 1000,
      "minimapCanvasOuterWidth": 58.8,
      "minimapCanvasOuterHeight": 400
    },
    "viewportColumn": 63,
    "isWordWrapMinified": false,
    "isViewportWrapping": false,
    "wrappingColumn": -1,
    "verticalScrollbarWidth": 14,
    "horizontalScrollbarHeight": 12,
    "overviewRuler": {
      "top": 0,
      "width": 14,
      "height": 400,
      "right": 0
    }
  },
  "wrappingInfo": {
    "isDominatedByLongLines": false,
    "isWordWrapMinified": false,
    "isViewportWrapping": false,
    "wrappingColumn": -1
  },
  "defaultColorDecorators": false,
  "colorDecoratorsActivatedOn": "clickAndHover",
  "inlineCompletionsAccessibilityVerbose": false
}
```

**Desktop (please complete the following information):**
 - OS: Any (macOS Monterey for browser, Ubuntu22.04 for server)
 - Browser: Chrome
 - Version: 4.6.0
   - monaco-editor (0.44.0)

**Additional context**

**Comment by LumaKernel** [comment]
After considering to switch to [react-monaco-editor](https://github.com/react-monaco-editor/react-monaco-editor), but I found some demerits of this alternative:

- It accesses widow.navigator, so unhappy with SSR
- It does not automatically resize based on container(or window) resize

I hope it is fixed. I'll try to find the time for investigating this too. Thanks. ✨

(I decided to use this library with monkey patch, because it is only for my private project for now)

**Comment by LumaKernel** [comment]
My monkey patch, note that it works only for my case.

```ts
const monkeyPatch = (editor: monacoEditor.IStandaloneCodeEditor) => {
  const origGetOption = editor.getOption.bind(editor);
  editor.getOption = (id: number) => {
    if (id > 16) {
      id -= 1;
    }
    // eslint-disable-next-line @typescript-eslint/no-unsafe-return
    return origGetOption(id);
  };
};
```

**Comment by voidranjer** [comment]
Hey @LumaKernel, I ran into the same issue and have looked into it.

Tldr; This bug is caused by a dependency version mismatch in the [monaco-loader](https://github.com/suren-atoyan/monaco-loader) repo instead of this one (monaco-react).

I wrote a detailed explanation and suggested some temporary (and permanent) fixes in the pull request, please see [Pull Request #43](https://github.com/suren-atoyan/monaco-loader/pull/43)

Hope it helps!
